transport/internal/tlsmux/listener.go

```diff
@@ -21,17 +21,22 @@
 package tlsmux
 
 import (
+	"context"
 	"crypto/tls"
 	"errors"
 	"net"
 	"sync"
+	"time"
 
 	"go.uber.org/net/metrics"
 	"go.uber.org/zap"
 )
 
 var (
 	errListenerClosed = errors.New("listener closed")
+
+	_sniffReadTimeout    = time.Second * 10
```

Comment:
if you have any reference how we come up those numbers, please link it here in the comments.

Reply:
Actually, these are experimental numbers and will be tuned later in production. gRPC sets a default timeout of 120s. I have added a TODO